sm4/src/x86/aesni.rs

```diff
@@ -0,0 +1,374 @@
+//! SM4 with X86 AES-NI instruction set
+//!
+//! Implementation was borrowed from <https://www.cnblogs.com/kentle/p/15826075.html> by kentle.
```

Comment:
I found an English description of the approach being used here which helped me understand how this actually works:

https://github.com/mjosaarinen/sm4ni

> The trick is to use affine transforms to emulate the SM4 S-Box with the AES S-Box. The S-Boxes are both based on finite field inversion, but use different affine transforms and even polynomial basis for the finite field. However, different polynomial bases are affine isomorphic.

Interesting approach!

Reply:
Yes it is.